torchci/clickhouse_queries/ttrs_percentiles/query.sql

```diff
@@ -159,88 +163,84 @@ workflow_failure AS (
     FIRST_VALUE(d.failure_time) OVER(
       PARTITION BY d.pr_number, d.sha, d.workflow_run_id
       ORDER BY d.failure_time
-    ) as failure_time,
+    ) as failure_time
   FROM
-    commit_job_durations d
+    commit_job_durations_steps d
 ),
 workflow_failure_buckets AS (
   SELECT
     -- When :one_bucket is set to true, we want the ttrs percentile over all the data
     DATE_TRUNC(
       'week',
       IF(
-        : one_bucket,
-        CURRENT_DATETIME(),
+        {one_bucket: Bool},
+        now(),
         start_time
       )
     ) AS bucket,
     *
   FROM
     workflow_failure
 ),
--- Within each bucket, figure out what percentile duration and num_commits each PR falls under
 percentiles AS (
   SELECT
-    bucket,
-    ttrs_mins,
-    workflow_url,
-    PERCENT_RANK() OVER(
-      PARTITION BY bucket
-      ORDER by
-        ttrs_mins
-    ) AS percentile,
-    sha,
-  FROM
-    workflow_failure_buckets
-),
--- Take the full list of percentiles and get just the ones we care about
-ttrs_percentile AS (
-  SELECT
-    p.bucket,
-    pd.percentile,
-    MIN(p.ttrs_mins) AS ttrs_mins
+    w.bucket,
+    quantileExact({percentile_to_get: Float32})(w.ttrs_mins) as custom,
+    quantileExact(.25)(w.ttrs_mins) as p25,
+    quantileExact(.5)(w.ttrs_mins) as p50,
+    quantileExact(.75)(w.ttrs_mins) as p75,
+    quantileExact(.9)(w.ttrs_mins) as p90
```

Comment:
CH has a quantile function, but specifying the quantile cannot use the value from a query, so instead I just define them all.  Unfortunately this means that the React code might need to be changed for the kpis page

Reply:
As long as it still shows the same data at the end.